app/javascript/react/components/ProductFormContainer.js

```diff
@@ -20,7 +44,15 @@ const ProductFormContainer = (props) => {
 
   const onSubmitHandeler = (event) => {
     event.preventDefault();
-    addNewProduct(productRecord);
+    if (validForSubmission()) {
+      addNewProduct(productRecord);
+    }
+    setProductRecord({
+      name: "",
+      description: "",
+      url: "",
+      image_url: ""
+    })
```

Comment:
This looks like it is always clearing the form when a user clicks submit for both a valid and invalid submission.  Lets go ahead and remove lines 50-55, because if there is an error and we stay on this page we want the form to remain filled out.  However in the case of a successful product submission and creation on the backend we will redirect to another page, so the form state wont matter anyways.